python-sdk/nuscenes/nuscenes.py

```diff
@@ -94,14 +94,12 @@ def __init__(self,
                 'Error: There are {} .bin files but {} lidarseg records.'.format(num_bin_files, num_lidarseg_recs)
             self.table_names.append('lidarseg')
 
-            lidarseg_categories = self.__load_table__('category_lidarseg')
-
             # Create mapping from class index to class name, and vice versa, for easy lookup later on.
             self.lidarseg_idx2name_mapping = dict()
             self.lidarseg_name2idx_mapping = dict()
-            for lidarseg_category in lidarseg_categories:
-                self.lidarseg_idx2name_mapping[lidarseg_category['index']] = lidarseg_category['label']
-                self.lidarseg_name2idx_mapping[lidarseg_category['label']] = lidarseg_category['index']
+            for lidarseg_category in self.category:
+                self.lidarseg_idx2name_mapping[lidarseg_category['index']] = lidarseg_category['name']
+                self.lidarseg_name2idx_mapping[lidarseg_category['name']] = lidarseg_category['index']
```

Comment:
Make sure the new (post-lidarseg) code throws an error if it sees the old version, e.g. if "index" does not exist.